### ISSUE TITLE
Security: Antispam ?

### ISSUE THREAD

**Issue by fredleger** [mailhog/MailHog#138]
I know this will be strange request. But any way to add a kind of spam user blocking ?

I can see differents ways :
* configurable delay after tcp connect
* rate limitating
* ip filtering (with white / black listing)
* DNS sanity check
* real anti spam plugable

Just sharing the thoughts here. Let's see what is easy to implement and what is not.

**Comment by kohenkatz** [comment]
I'm curious what the use-case is that requires this to be added, given that MailHog isn't really designed to be exposed to the public - it's supposed to be for your own personal use in development so you can test your code that sends emails.

**Comment by fredleger** [comment]
we were looking to a solution to receive e-mails to dynamic adresses (not known in advance like qsdhqsjkhjsqhdqsdjqsdhqskjh33668228@exemple.com) and retrieve e-mails from the API. After doing some research we didn't found any solutions doing that (except hosted services but we want to host ourselves) so we endded up using maihug who actually fit the need so far. But in deed it lacks some security features.

If you have any proposals we are up to ...

**Comment by ian-kent** [comment]
Hi @fredleger - it's not really meant for this - it's purely a testing tool, as @kohenkatz points out its not designed to be exposed as a public server (and would be relatively dangerous to do so!)

Can't really suggest any alternatives - postfix/exim etc are fairly customisable, but also an unwanted headache if you don't have the time/experience to configure/maintain, and don't come with an API built-in that I know of.

I'd love to turn MailHog into a real mail server at some point, but lack of time and competing projects makes it unlikely.

Having said that, I did start putting [MailHog-MTA](https://github.com/mailhog/MailHog-MTA) and [MailHog-IMAP](https://github.com/mailhog/MailHog-IMAP) together, with the goal of being a production-ready distributed mail server, but got distracted in the middle of implementing POP3. Turns out ancient email RFCs are a nightmare to implement!
